sdk/eventhub/azure-eventhub/samples/async_samples/iot_hub_connection_string_receive_async.py

```diff
@@ -48,10 +49,10 @@ def generate_sas_token(uri, policy, key, expiry=None):
     return 'SharedAccessSignature ' + urlencode(result)
 
 
-def parse_iot_conn_str(iothub_conn_str):
-    hostname = None
-    shared_access_key_name = None
-    shared_access_key = None
+def parse_iot_conn_str(iothub_conn_str) -> Tuple[str,str,str]:
+    hostname = ''
```

Comment:
Using empty string here avoids allows us to avoid it being typed as Optional[str] (ie, before we used None). But I'm not sure if this is a good way to do it, or if there is a better way.

We have a check (line 64) that already looks to see if the value is not None (or empty string, for that matter) so, at runtime, we do guarantee that the value will never be None. However, the Python typechecker didn't know that, so we had to return a Tuple[Optional[str], Optional[str], Optional[str]] instead, which isn't great.

Is the approach here correct in a "this is how we'd do it in the Python SDK?"

Reply:
OOB discussion with Kashif: We generally initialize as None if they haven't been set and no other default value makes sense. In this case, if any of these are empty strings, this will raise an error anyway. So `''` doesn't really make sense here.

It's not ideal, but for the sake of typing + returning `Tuple[str, str, str]`, we could potentially cast these all as strings? or to fully remove 52-54, we could also add all keys and values to a `Dict[str, str]` and do something like:
```
try:
    hostname, shared_access_key_name, shared_access_key = vals['hostname'], vals['sharedaccesskeyname'], vals['sharedaccesskey']
except KeyError:
    raise ValueError("Invalid conn str")
return hostname, shared_access_key_name, shared_access_key
```

Reply:
If _you're_ interested in changing/trying out a different approach, that sounds great! O/w I can update this PR/fix it in a separate typing PR. 🙂

Can't promise the badge, but I'll give you permission to use this emoji 🐍